### PR TITLE
Add machine autoscaler modal

### PR DESCRIPTION
**Pull request by spadgett**
@derekwaynecarr @smarterclayton 
/hold

Screen Shot 2019-05-01 at 2 06 11 PM

Screen Shot 2019-05-01 at 2 19 10 PM

Screen Shot 2019-05-01 at 2 19 30 PM

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
please take a look @enxebre @frobware @bison 

makes it easier to demonstrate feature/function versus hand crafted yaml.

File: frontend/public/models/yaml-templates.ts [hunk 1]
```diff
@@ -797,4 +797,17 @@ spec:
   machineSelector:
     matchLabels:
       node-role.kubernetes.io/master: ""
+`).setIn([referenceForModel(k8sModels.MachineAutoscalerModel), 'default'], `
+apiVersion: "autoscaling.openshift.io/v1beta1"
+kind: "MachineAutoscaler"
+metadata:
+  name: "worker-us-east-1a"
+  namespace: "openshift-cluster-api"
```
Comment (bison) [inline frontend/public/models/yaml-templates.ts]: This is the wrong namespace. It should be: `openshift-machine-api`

Comment (spadgett) [reply frontend/public/models/yaml-templates.ts]: Thanks, I updated it. It gets replaced anyway at runtime anyway FWIW. This is a template for the YAML editor, which defaults the namespace to the current namespace. It's not used for the dialog.

Comment (bison) [reply frontend/public/models/yaml-templates.ts]: Cool. Thought that might be the case, but wanted to mention it anyway. Thanks!

**Comment by spadgett** [comment]
/hold cancel
Targeting to 4.2

**Comment by spadgett** [comment]
#1531 should fix the OLM test failure.

/retest

File: frontend/public/components/machine-autoscaler.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+const { common } = Kebab.factory;
+const menuActions = [...common];
+const machineAutoscalerReference = referenceForModel(MachineAutoscalerModel);
+
+const MachineAutoscalerTargetLink: React.FC<MachineAutoscalerTargetLinkProps> = ({obj}) => {
+  const targetAPIVersion: string = _.get(obj, 'spec.scaleTargetRef.apiVersion');
+  const targetKind: string = _.get(obj, 'spec.scaleTargetRef.kind');
+  const targetName: string = _.get(obj, 'spec.scaleTargetRef.name');
+  if (!targetAPIVersion || !targetKind || !targetName) {
+    return <React.Fragment>-</React.Fragment>;
```
Comment (TheRealJon) [inline frontend/public/components/machine-autoscaler.tsx]: Can this just be:

```jsx
return '-';
```

Comment (spadgett) [reply frontend/public/components/machine-autoscaler.tsx]: TypeScript doesn't like it :(

Comment (TheRealJon) [reply frontend/public/components/machine-autoscaler.tsx]: Hmm, okay. 🤷‍♂

Comment (spadgett) [reply frontend/public/components/machine-autoscaler.tsx]: Yeah, I think it's a problem with the React type definitions, but `React.Fragment` works

```
ERROR in /Users/sam/go/src/github.com/openshift/console/frontend/public/components/machine-autoscaler.tsx(21,7):
TS2322: Type '({ obj }: PropsWithChildren<MachineAutoscalerTargetLinkProps>) => Element | "-"' is not assignable to type 'FunctionComponent<MachineAutoscalerTargetLinkProps>'.
  Type 'Element | "-"' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>'.
    Type '"-"' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>'.
```

**Comment by spadgett** [comment]
Rebased

File: frontend/public/components/machine-autoscaler.tsx (hunk 2 above)
Comment (TheRealJon) [reply frontend/public/components/machine-autoscaler.tsx]: Seems to be a known issue. https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20544

**Pull request merged**